src/Shaders/ShadersInclude/lightFragment.fx

```diff
@@ -162,6 +162,29 @@
         #else
             #if defined(SHADOWCUBE{X})
                 shadow = computeShadowCube(light{X}.vLightData.xyz, shadowSampler{X}, light{X}.shadowsInfo.x, light{X}.depthValues);
+			#elif defined(SHADOWCSM{X})
+				int index = -1;
+				for (int i = 0; i < num_cascades{X}; i++) {
+					if (vPositionFromCameraCSM{X}.z <= ViewFrustumZCSM{X}[i]){
+						index = i;
+						break;
+					}
+				}
+				if (index == 0){

```

Comment:
If we lock down CSM to webgl2 is it possible to access through dynamic index in the array ?

Reply:
Yh, same thoughts, these codes were initially super easy and short, like so:
```
shadow = computeShadow(vPositionFromLightCSM{X}[i], vDepthMetricCSM{X}[i], shadowSamplerArrayCSM{X}[i], light{X}.shadowsInfo.x, light{X}.shadowsInfo.w);
```
Guess what ?
```
BJS - [15:24:32]: Error: C:\fakepath(157,12-35): warning X3550: sampler array index must be a literal expression, forcing loop to unroll
```
plus a whole bunch of errors. I guess this (https://www.khronos.org/opengl/wiki/Data_Type_(GLSL)#Opaque_arrays) is in effect.

Reply:
Is it the same if you manually unroll ?

As we do for lights actually.

Reply:
I've tried ```#pragma unroll 4``` (and whatever other syntax) before the loop prior to PR, same result. Also tried searching the repo for pragma or unroll, but I don't see it at all. Unless you are not doing it this way, then I need an example.

Reply:
We do it for light via special macros that are expanded before compilation.

#include<lightFragment>[0..maxSimultaneousLights]